### ISSUE TITLE
[FEATURE-REQUEST]

### ISSUE THREAD

**Issue by shexuan** [vaexio/vaex#956]
**Description**
Could you please supply a function like pandas `explode` to  unnest a column to rows?

**Is your feature request related to a problem? Please describe.**
I want to make a vocabulary index of each item in a list column, examples:
```
|column|
|['a', 'b', 'c']|
|['d', 'e']|
```
So Firstly I need to get all the words,  then the vocabulary would be {'a':0, 'b':1, 'c':2, 'd':3, 'e':4}。

**Additional context**
Alternatively, Is there a function like pandas `iterrows`, so I could easily build a vocabulary.

**Comment by JovanVeljanoski** [comment]
Hi, 

Please check out this issue #952 for a brief discussion on supported types.
As discussed there, non primitive types are officially not supported (so things like lists, stricts etc..).  We are slowly moving to adopting arrow, so the next release (coming soon) will get a step closer to that direction.

As we adopt Arrow more, methods such as explode will be implemented - perhaps Arrow already has something like this, or is in their roadmap. 

As for iterating over rows: check out [the documentation](https://docs.vaex.io/en/latest/example_io.html#In-memory-data-representation): methods like `.to_pandas`, `.to_dict`, .`to_arrays` etc all implement the `chunk_size` argument and do what you want.

Finally: our proposed workflow (at the moment at least) is that one can use vaex and pandas together. Pandas to get the data into the right format (so things like, explode, pivot, melt, etc..) that mostly require all the data to be in memory, since they are doing some kind of fundamental reshaping. After that the data is passed to vaex for faster calculations. They packages do not need to be mutually exclusive.

Let's keep this issue open to track the progress of the `explode` method. 

Thanks!

**Comment by maartenbreddels** [comment]
Yes, explode is an interesting one, I think we could do this 'half' lazily. But I'd give it low priority for the moment.
